src/nvim/screen.c

```diff
@@ -665,19 +675,21 @@ static void win_update(win_T *wp)
 
   type = wp->w_redr_type;
 
+  win_grid_alloc(wp, true);
```

Comment:
I'm not sure we want to default to clear window grids on window resize. It is better to define what happens with the remaining content on resize, to reduce redrawing when a separator is dragged, etc. It looks like the present code is prepared for preserving the bottom line (for gvim to behave like a terminal when "needed", but I don't think nvim currently uses it, it just refrains from resizing the grid in "press RETURN" state). But perhaps it would be simpler (for new GUIs) to preserve the topline as it is defined as the origin `(0,0)`.